tests/test_backends.py

```diff
@@ -0,0 +1,142 @@
+import pytest
+import numpy as np
+
+from opt_einsum import contract, helpers, contract_expression, backends
+
+try:
+    import tensorflow as tf
+    found_tensorflow = True
+except ImportError:
+    found_tensorflow = False
+
+try:
+    import os
+    os.environ['MKL_THREADING_LAYER'] = 'GNU'
+    import theano
+    found_theano = True
+except ImportError:
+    found_theano = False
+
+try:
+    import cupy
+    found_cupy = True
+except ImportError:
+    found_cupy = False
+
+try:
+    import dask.array as da
+    found_dask = True
+except ImportError:
+    found_dask = False
+
+try:
+    import sparse
+    found_sparse = True
+except ImportError:
+    found_sparse = False
+
+
+tests = [
```

Comment:
Can we actually run these test through the normal test set?

Reply:
Yes, they are all on travis (I fixed an error with ``dask`` earlier), apart from ``cupy`` which actually requires CUDA / a  GPU to run.